uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -167,6 +194,37 @@ define([
         },
 
         /**
+         * Sets the flag to mark file for deletion. The file is not yet deleted
+         * from the server.
+         */
+        deleteAttachment: function (attachmentID) {
+          this.files = this.files.map(function (file) {
```

Comment:
The logic is still different between `comments` and `attachments`. We should definitely have similar same approach for both.

I prefer comments approach, because there we dont to need to search through an array to fetch the object, instead the object is passed through parameter.